uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1025.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1025 {
+
+  /**
+   * Removes Instant Messenger options
+   */
+  public function upgrade_1025() {
+    $this->up1025_removeInstantMessengerOptions([
+      'Yahoo',
+      'MSN',
+      'AIM',
+      'Jabber',
+    ]);
+
+    return TRUE;
+  }
+
+  /**
+   * Removes some instant messenger options if not used, and disable it if used
+   */
+  private function up1025_removeInstantMessengerOptions($accounts) {
```

Comment:
This function does not only remove Instant messenger options only
This function actually does two things:
(1) It deletes accounts not in use 
(2) It disables if they are in use.
I would suggest to:
Rename function to deleteSocialAccountsNotInUseAndDisableThoseInUse or something better than this.
To make it even clearer, extract the logic for deleting the account and disabling the account into two mini separate functions and call these functions from within the foreach.